v2/pkg/engine/datasource/httpclient/nethttpclient.go

```diff
@@ -293,14 +308,13 @@ func DoMultipartForm(
 		}
 		hasWrittenFileName = true
 
-		fmt.Fprintf(fileMap, `"%d":["%s"]`, i, file.variablePath)
+		_, _ = fmt.Fprintf(fileMap, `"%d":["%s"]`, i, file.variablePath)
 
 		key := fmt.Sprintf("%d", i)
-		_, _ = h.WriteString(file.Path())
 		temporaryFile, err := os.Open(file.Path())
 		tempFiles = append(tempFiles, temporaryFile)
 		if err != nil {
```

Comment:
`defer` below only happens when there was no early return before. If for some file we return early, and some fiels were previsouly added to `tempFiles` then we will not remove close/remove those files. Also the if check should happen before adding the temporaryFile to the slice.